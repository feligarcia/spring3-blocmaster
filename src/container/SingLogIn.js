import React from "react";
import Login from "../components/Login";
import styled from "styled-components";
import Logo from "../data/images/logo.png";
import { useNavigate } from "react-router-dom";
import SignIn from "../components/SignIn";
import { useSelector } from "react-redux";

const DivSingLogin = styled.section`
  margin: 5% auto;
  padding: 0 5%;
  display: flex;
  flex-direction: column;
  min-width: 425px;
  max-width: 500px;
`;

const LogoNav = styled.img`
  height: 84px;
  width: fit-content;
  background-color: inherit;
  align-self: center;
`;

const SingLogIn = () => {
  const { registroMostrar } = useSelector((store) => store.app);
  const navigate = useNavigate();
  return (
    <DivSingLogin>
      <LogoNav src={Logo} onClick={() => navigate("/")}></LogoNav>
      <br></br>
      <h2>Registrar nuevo usuario</h2>
      {registroMostrar ? (
        <Login />
      ) : (
        
        <SignIn />
      )}
    </DivSingLogin>
  );
};

export default SingLogIn;
