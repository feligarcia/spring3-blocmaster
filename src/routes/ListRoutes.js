import { Routes, Route, Navigate } from "react-router-dom";
import { ModalMovie } from "../components/ModalMovie";
import Home from "../container/Home";
import Personal from "../container/Personal";

export const ListRoutes = () => {
  return (
    <div>
      <Routes>
        <Route path="/" element={<Home />}>
        </Route>
        <Route path="modal/:id" element={<ModalMovie />} />
        <Route path="personal" element={<Personal />} />
        <Route path="*" element={<Navigate to="/" />} />
      </Routes>
    </div>
  );
};
