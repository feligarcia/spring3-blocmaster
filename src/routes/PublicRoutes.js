
import { Navigate } from 'react-router-dom';

const  PublicRoutes = ({isAuthenticated, children})=> {
  
   
  
    return !isAuthenticated
    ? children
    : <Navigate to='/*' />
}

export default PublicRoutes
