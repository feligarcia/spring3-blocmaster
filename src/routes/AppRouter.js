import React, { useEffect, useState } from "react";
import { Route, Routes, HashRouter } from "react-router-dom";
import { getAuth, onAuthStateChanged } from "firebase/auth";
import SingLogIn from "../container/SingLogIn";
import PublicRoutes from "./PublicRoutes";
import { ListRoutes } from "./ListRoutes";
import { PrivateRoute } from "./PrivateRoutes";
import Loader from "../components/Loader";

// import CargarMoviesAPI from "../components/CargarMoviesAPI";

function AppRouter() {
  const [checking, setChecking] = useState(true);
  const [isLogin, setIsLogin] = useState(false);

  useEffect(() => {
    const auth = getAuth();
    onAuthStateChanged(auth, (user) => {
      if (user?.uid) {
        setIsLogin(true);
        const {displayName, email, photoURL} = user
        const sesion = {displayName,email,photoURL}
        localStorage.setItem('userBMApp', JSON.stringify(sesion))
      } else {
        setIsLogin(false);
        localStorage.removeItem('userBMApp')
      }
      setChecking(false);
    });
  }, [setChecking, setIsLogin]);

  if (checking) {
    return <Loader />;
  }

  return (
    <div>
      <HashRouter>
        <Routes>
          <Route
            path="/login"
            element={
              <PublicRoutes isAuthenticated={isLogin}>
                {/* <CargarMoviesAPI /> */}
                <SingLogIn />
              </PublicRoutes>
            }
          />

          <Route
            path="/registro"
            element={
              <PublicRoutes isAuthenticated={isLogin}>
                <SingLogIn />
              </PublicRoutes>
            }
          />

          <Route
            path="/*"
            element={
              <PrivateRoute isAuthenticated={isLogin}>
                <ListRoutes />
              </PrivateRoute>
            }
          />
        </Routes>
      </HashRouter>
    </div>
  );
}

export default AppRouter;
