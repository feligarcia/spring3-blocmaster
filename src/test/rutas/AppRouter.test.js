import React from "react";
import "@testing-library/jest-dom";
import "../../setupTest";
import { shallow } from "enzyme";
import { PrivateRoute } from "../../routes/PrivateRoutes";
import  PublicRoutes  from "../../routes/PublicRoutes";
import { ListRoutes } from "../../routes/ListRoutes";
import SingLogIn from "../../container/SingLogIn";

describe("Validar Rutas Publicas y Privadas", () => {
  test("Validar render de <PrivateRoute />", () => {
    const isLogin = false;
    const wrapper = shallow(
      <PrivateRoute isAuthenticated={isLogin} children={<ListRoutes />} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test("Validar render de <PublicRoutes />", () => {
    const isLogin = false;
    const wrapper = shallow(
        <PublicRoutes isAuthenticated={isLogin} children={<SingLogIn />} />
      );
    expect(wrapper).toMatchSnapshot();
  });
});
