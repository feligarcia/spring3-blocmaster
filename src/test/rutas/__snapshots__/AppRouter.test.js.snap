// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validar Rutas Publicas y Privadas Validar render de <PrivateRoute /> 1`] = `<ListRoutes />`;

exports[`Validar Rutas Publicas y Privadas Validar render de <PublicRoutes /> 1`] = `
<Navigate
  to="/*"
/>
`;
